app/hotels/src/HotelsFilterContext.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import * as React from 'react';
+
+import type {
+  FilterParams,
+  ActiveFilters,
+  OnChangeFilterParams,
+} from './filter/FilterParametersType';
+
+const defaultFilterParams = {
+  starsRating: [],
+  minPrice: null,
+  maxPrice: null,
+  freeCancellation: false,
+  hotelFacilities: [],
+  minScore: null,
+};
+
+const defaultActiveFilters = {
+  isPriceFilterActive: false,
+  isStarsFilterActive: false,
+  isMinScoreActive: false,
+  isHotelFacilitiesActive: false,
+};
+
+const { Consumer, Provider: ContextProvider } = React.createContext({
+  filterParams: defaultFilterParams,
+  activeFilters: defaultActiveFilters,
+  actions: {
+    setFilter: () => {},
+  },
+});
+
+type Props = {|
+  children: React.Node,
+|};
+
+type State = {|
+  filterParams: FilterParams,
+  activeFilters: ActiveFilters,
+  actions: {|
+    setFilter: OnChangeFilterParams => void,
+  |},
+|};
+
+class Provider extends React.Component<Props, State> {
+  setFilter = (action: OnChangeFilterParams) => {
+    const newState = {
+      ...this.state,
+      filterParams: {
```

Comment:
Shouldn't you be using functional set state here?

Reply:
Yes, fixed - please check it again.